Documentation/Assignment 2/tex/assign_2_functional.tex

```diff
@@ -0,0 +1,46 @@
+\chapter{Functional Requirements}
+
+\section{Introduction}
+In this document the requirements for assignment 2 will be stated. There will be improvements to the code, as well as new functionalities. \noindent This will be done using the \fnurl{MoSCoW Model}{https://en.wikipedia.org/wiki/MoSCoW_method}.
+
+\section{Must Have}
+\begin{itemize}
+\itemsep0em
+\item The player must be able to fall through the floor.
+\item The player must reappear after falling through the floor at the corresponding square in the ceiling.
+\item The monsters must be able to fall though the floor.
+\item The monsters must reappear after falling through the floor at the corresponding square in the ceiling.
+\item The monsters must be able to jump.
+\item The monsters must be able to jump and move around randomly.
+\item The player must be able to jump in a more natural way, so a decrease in speed when the player jumps.
+\item The player must be able to fall in a more natural way, so an increase in speed when the player falls.
+\item The monsters must be able to jump in a more natural way, so a decrease in speed when the monster jumps.
+\item The monsters must be able to fall in a more natural way, so an increase in speed when the monster falls.
+\item The bubbles must be able to float to the top of the screen.
+\item The bubbles must disappear after a period of time.
+\item The player must be able to jump on a bubble, and use it like a lift.
+\end{itemize}
+
+\section{Should Have}
+\begin{itemize}
+\itemsep0em
+\item The bubbles should be able to float through the ceilings, but stop at the top ceiling.
```

Comment:
"The bubbles should be able to float through the platforms, but should stop at the ceiling."